### PR TITLE
Reset Button

### PR DESCRIPTION
**Pull request by suchoX**
1. Added Reset Button in Barricade Activity
2. Added Delay reset in Barricade

### CODE REVIEW

File: barricade/src/main/java/com/mutualmobile/barricade/activity/BarricadeActivity.java [hunk 1]
```diff
@@ -116,4 +121,19 @@ private void showEditDialog() {
     EditGlobalDelayDialogFragment dialogFragment = new EditGlobalDelayDialogFragment();
     dialogFragment.show(fm, EditGlobalDelayDialogFragment.class.getName());
   }
+
+  private void showResetDialog() {
+    AlertDialog.Builder builder = new AlertDialog.Builder(this);
+    builder.setMessage(getString(R.string.reset_message))
+        .setCancelable(true)
+        .setNeutralButton(R.string.ok, new DialogInterface.OnClickListener() {
```
Comment (mustafa01ali) [inline barricade/src/main/java/com/mutualmobile/barricade/activity/BarricadeActivity.java]: Replace with positive and negative buttons

File: barricade/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -2,8 +2,11 @@
   <string name="app_name">Barricade</string>
   <string name="title_activity">Barricade Config</string>
   <string name="menu_delay">Delay</string>
+  <string name="menu_reset">Reset</string>
   <string name="global_delay_ms">Global delay (ms)</string>
   <string name="title_edit_global_delay">Edit global delay (ms)</string>
   <string name="required">Required</string>
   <string name="updated">Updated</string>
+  <string name="reset_message">Are You sure you want to reset settings?</string>
```
Comment (mustafa01ali) [inline barricade/src/main/res/values/strings.xml]: Replace with `Are you sure you want to reset all changes?`

File: barricade/src/main/res/menu/main.xml [hunk 3]
```diff
@@ -6,4 +6,10 @@
       android:icon="@drawable/ic_timer_white"
       app:showAsAction="ifRoom"
       android:title="@string/menu_delay"/>
+
+  <item
+      android:id="@+id/menu_reset"
+      android:icon="@drawable/ic_reset"
+      app:showAsAction="ifRoom"
+      android:title="@string/menu_delay"/>
```
Comment (mustafa01ali) [inline barricade/src/main/res/menu/main.xml]: Add a proper title

File: barricade/src/main/res/drawable/ic_reset.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (mustafa01ali) [inline barricade/src/main/res/drawable/ic_reset.xml]: This icon is misleading, replace with https://material.io/icons/#ic_undo

File: barricade/src/main/java/com/mutualmobile/barricade/activity/BarricadeActivity.java [hunk 5]
```diff
@@ -107,6 +109,9 @@ private void setResponsesView(String endpoint) {
     } else if (itemId == R.id.menu_delay) {
       showEditDialog();
       return true;
+    } else if (itemId == R.id.menu_reset) {
```
Comment (tushar-nallan) [inline barricade/src/main/java/com/mutualmobile/barricade/activity/BarricadeActivity.java:112]: Use switch case

File: barricade/src/main/java/com/mutualmobile/barricade/activity/BarricadeActivity.java [hunk 6]
```diff
@@ -116,4 +121,19 @@ private void showEditDialog() {
     EditGlobalDelayDialogFragment dialogFragment = new EditGlobalDelayDialogFragment();
     dialogFragment.show(fm, EditGlobalDelayDialogFragment.class.getName());
   }
+
+  private void showResetDialog() {
+    AlertDialog.Builder builder = new AlertDialog.Builder(this);
```
Comment (tushar-nallan) [inline barricade/src/main/java/com/mutualmobile/barricade/activity/BarricadeActivity.java]: Continue the builder. Also .show() internally calls .create() so no need of that

File: barricade/src/main/res/drawable/ic_reset.xml (hunk 4 above)
Comment (tushar-nallan) [inline barricade/src/main/res/drawable/ic_reset.xml]: @mustafa01ali Do we want to make it mandatory for the user of this library to have vector support or should we use image drawables? Currently we have 2 vector drawables - red tick mark, and timer icon.

**Review by tushar-nallan — APPROVED** [review]

File: barricade/src/main/res/drawable/ic_reset.xml (hunk 4 above)
Comment (mustafa01ali) [reply barricade/src/main/res/drawable/ic_reset.xml]: Yes, every app should ideally adopt vector drawables at this point. If we can push folks to adopt it, why not.

File: barricade/src/main/java/com/mutualmobile/barricade/activity/BarricadeActivity.java (hunk 5 above)
Comment (suchoX) [reply barricade/src/main/java/com/mutualmobile/barricade/activity/BarricadeActivity.java:112]: Switch Case doesnt work for resource variables in Library modules

**Review by mustafa01ali — CHANGES_REQUESTED** [review]

**Review by mustafa01ali — APPROVED** [review]

**Pull request merged**